lib/resource.js

```diff
@@ -1,9 +1,20 @@
 import types from './config/resource-types.js';
+import crypto from 'crypto';
+import fs from 'fs/promises';
+import path from 'path';
+import { compress, decompress } from './utils/index.js';
 
 class Resource {
-	constructor (url, filename) {
-		this.url = url;
-		this.filename = filename;
+	constructor (url, filename, tempMode, tempDir) {
+		this.tempMode = tempMode || 'memory';
+		this.tempDir = tempDir;
+
+		if (this.tempMode === 'filesystem' && !this.tempDir) {
```

Comment:
It looks like `tempDir` is already set in `lib/scraper.js` so we will have it defined at this point